src/javascript/app_2/App/Components/Form/TimePicker/time_picker.jsx

```diff
@@ -0,0 +1,123 @@
+import { observer }       from 'mobx-react';
+import PropTypes          from 'prop-types';
+import React              from 'react';
+import TimePickerDropdown from './time_picker_dropdown.jsx';
+import InputField         from '../input_field.jsx';
+
+class TimePicker extends React.PureComponent {
+    constructor (props) {
+        super(props);
+        this.state = {
+            is_open: false,
+            value  : '',
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    toggleDropDown = () => {
+        this.setState({ is_open: !this.state.is_open });
+    };
+
+    handleChange = (arg) => {
+        // To handle nativepicker;
+        const value = typeof arg === 'object' ? arg.target.value : arg;
+
+        if (value !== this.props.value) {
+            this.props.onChange({ target: { name: this.props.name, value } });
+        }
+    };
+
+    saveRef = (node) => {
+        if (!node) return;
+        if (node.nodeName === 'INPUT') {
+            this.target_element = node;
+            return;
+        }
+        this.wrapper_ref = node;
+    };
+
+    handleClickOutside = (event) => {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target)) {
+            if (this.state.is_open) {
+                this.setState({ is_open: false });
+            }
+        }
+    };
+
+    render() {
+        const prefix_class = 'time-picker';
+        const {
+            value,
+            name,
+            is_nativepicker,
+            placeholder,
+            start_time,
+            end_time,
+            validation_errors,
+        } = this.props;
+        return (
+            <div
+                ref={this.saveRef}
+                className={`${prefix_class}${this.props.padding ? ' padding' : ''}${this.state.is_open ? ' active' : ''}`}
+            >
+                {
+                    is_nativepicker
+                        ? <input
+                            type='time'
+                            id={`${prefix_class}-input`}
+                            value={value}
+                            onChange={this.handleChange}
+                            name={name}
+                            min={start_time}
+                            max={end_time}
+                        />
+                        : (
+                            <React.Fragment>
+                                <InputField
+                                    error_messages={validation_errors}
+                                    type='text'
+                                    is_read_only
+                                    id={`${prefix_class}-input`}
+                                    className={`${prefix_class}-input ${this.state.is_open ? 'active' : ''}`}
```

Comment:
Nitpick: We can use `classNames` library to make this cleaner and more readable instead of string template.

Reply:
Agree. I was planning to use that too but I hold it off until I can work the popover class, so that timepicker and datepicker don't need 2 different implementation to do the same thing.

Reply:
Not sure what do you mean by two different implementations? I don't see the relevance in that with how we should refactor the `classNames`.

Reply:
I suppose I should treat this as a one off thing just in case I don't get the chance to work on what I wanted to.

Reply:
The `Popover` component just clones the child elements/component and wraps itself around as a parent `component`, you will still need to refactor the `classNames` for the child components used. 
Either way, it's up to you whether you refactor it now or later on a supplementary card. 

I just assumed since the PR is called `Refactor/Timepicker`, it is more relevant here.